components/studentPortal/layout/Sidebar.tsx

```diff
@@ -74,6 +79,18 @@ export const Sidebar: React.FC = () => {
     }
   }, [router.pathname]);
 
+  useEffect(() => {
+    if (user) {
+      fetchProfilePicture(user.uid).then((imageUrl) => {
```

Comment:
as a frontend developer i should be able to fetch the profile picture for the current logged in user easily and not have to worry about where the image is coming from. 

if we had more complex backend systems and needed to migrate from AWS to Google Cloud, this front end code shouldn't need to change.

Reply:
Awesome! Ensuring the front-end code remains agnostic to back-end changes is a brilliant way to guarantee our system's flexibility in the long run.